src/connector/src/source/cdc/external/sql_server.rs

```diff
@@ -169,35 +171,33 @@ impl SqlServerExternalTable {
     }
 }
 
-fn type_to_rw_type(col_type: &ColumnType) -> ConnectorResult<DataType> {
-    let dtype = match col_type {
-        ColumnType::Bit => DataType::Boolean,
-        ColumnType::Bitn => DataType::Bytea,
-        ColumnType::Int1 => DataType::Int16,
-        ColumnType::Int2 => DataType::Int16,
-        ColumnType::Int4 => DataType::Int32,
-        ColumnType::Int8 => DataType::Int64,
-        ColumnType::Float4 => DataType::Float32,
-        ColumnType::Float8 => DataType::Float64,
-        ColumnType::Decimaln | ColumnType::Numericn => DataType::Decimal,
-        ColumnType::Daten => DataType::Date,
-        ColumnType::Timen => DataType::Time,
-        ColumnType::Datetime
-        | ColumnType::Datetimen
-        | ColumnType::Datetime2
-        | ColumnType::Datetime4 => DataType::Timestamp,
-        ColumnType::DatetimeOffsetn => DataType::Timestamptz,
-        ColumnType::NVarchar | ColumnType::NChar | ColumnType::NText | ColumnType::Text => {
-            DataType::Varchar
-        }
-        // Null, Guid, Image, Money, Money4, Intn, Bitn, Floatn, Xml, Udt, SSVariant, BigVarBin, BigVarChar, BigBinary, BigChar
+fn type_to_rw_type(col_type: &str) -> ConnectorResult<DataType> {
+    let dtype = match col_type.to_lowercase().as_str() {
+        "bit" => DataType::Boolean,
+        "binary" | "varbinary" => DataType::Bytea,
+        "tinyint" | "smallint" => DataType::Int16,
+        "integer" | "int" => DataType::Int32,
+        "bigint" => DataType::Int64,
+        "real" => DataType::Float32,
+        "float" => DataType::Float64,
+        "decimal" | "numeric" => DataType::Decimal,
+        "date" => DataType::Date,
+        "time" => DataType::Time,
+        "datetime" | "datetime2" | "smalldatetime" => DataType::Timestamp,
+        "datetimeoffset" => DataType::Timestamptz,
+        "char" | "nchar" | "varchar" | "nvarchar" | "text" | "ntext" | "xml"
+        | "uniqueidentifier" => DataType::Varchar,
         mssql_type => {
             // NOTES: user-defined enum type is classified as `Unknown`
             tracing::warn!(
                 "Unknown Sql Server data type: {:?}, map to varchar",
                 mssql_type
             );
-            DataType::Varchar
+            return Err(anyhow!(
+                "Unknown Sql Server data type: {:?}, map to varchar",
```

Comment:
It seems we won't map these "unknown" types to varchar, because an error is returned here.

Reply:
Thanks! Updated.